### DISCUSSION TITLE
How to run a Micro locally

### DISCUSSION THREAD

**Discussion by MystPi** [deta/cloud-docs#232] [Help] [answered]
Say I have this code:
```js
const express = require('express');
const app = express();

app.get('/', (req, res) => {
  res.send('Hello World!');
});

module.exports = app;
```
I could add `app.listen()` to the end of the file to test it locally, but I would have to delete it each time I deployed the micro. Is there any other way to run micros locally?

**Comment by abdelhai** [comment, marked as answer]
you can have `app.listen(...)` at the end of you file, this won't cause any problem. 
if you must use a clean solution, check if you're outside a Micro first. this code should work:

```js
module.exports = app;
!process.env.DETA_RUNTIME && app.listen(3000):
```

**Reply by MystPi** [reply]
Okay, thanks!
